federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -184,6 +187,14 @@ func (cc *ClusterController) UpdateClusterStatus() error {
 				}
 			}
 		}
+		clusterClientSet, _ := NewClusterClientset(&cluster)
+		zones, region, err := GetClusterZones(clusterClientSet)
+		if err != nil {
+			glog.Errorf("Failed to get zones and region for cluster %s: %v", cluster.Name, err)
```

Comment:
Actually, looking more closely, line 202 falls into the same category.  We don't want to return an error there either, as we want to continue processing remaining clusters.  There is a sync loop around this (currently configured to a 40 second period), which will retry this all again later.